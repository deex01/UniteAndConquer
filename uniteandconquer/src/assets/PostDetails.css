.post-details-page {
  height: 100%;
  display: grid;
  grid-template-columns: 0.4fr 0.6fr;
}

.desired-quantity {
  max-width: 330px;
}

.item-name {
  font-size: 25px;
}

.post-details-content {
  display: grid;
  grid-template-columns: 0.5fr 0.5fr;
}

ul {
  padding: 0;
}

li {
  display: grid;
  grid-template-columns: 0.3fr 0.4fr;
  max-width: 400px;
  font-size: 19px;
  margin: 15px;
}

.size {
  max-width: 400px;
  font-size: 19px;
  margin: 5px;
}

.group-members {
  text-align: left;
}

.item-detail {
  max-width: 400px;
  font-size: 19px;
  margin: 15px;
}

.post-detail-tags {
  margin-top: 30px;
  text-align: left;
}

.tags-container {
  display: grid;
  grid-template-columns: 0.25fr 0.25fr 0.25fr 0.25fr;
  width: 60%;
  margin-top: 10px;
}

.tags-label {
  font-weight: bold;
}

.buttons-container {
  margin: auto;
  width: 52%;
  margin-bottom: 40px;
}

.comment-section {
  padding: 15px;
}