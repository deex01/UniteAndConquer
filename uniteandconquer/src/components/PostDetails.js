/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import Sidebar from './Sidebar';
import Comment from './Comment';
import '../assets/PostDetails.css';

const PostDB = require('../modules/PostDB');

// In the post details page, one can choose to join the group
// if successful, redirect the user to the post which the user joins in
function PostDetails() {
  const [desiredQuantity, setDesiredQuantity] = useState(0);
  const navigate = useNavigate();
  const myStorage = window.sessionStorage;
  const joinGroup = () => {
    const userID = myStorage.getItem('UserID');
    const postID = 0;
    const quantity = 2;
    PostDB.joinGroup(userID, postID, quantity, (success, err) => {
      if (success) {
        navigate(`/post-details${postID}`);
      } else {
        console.log(err);
      }
    });
  };
  const back = () => {
    navigate('/');
  };

  return (
    <div className="post-details-page">
      {Sidebar()}
      <div className="post-details-container">
        <div className="item-name">
          <h1>Item Name</h1>
        </div>
        <div className="post-details-content">
          <div className="item-details">
            <h3>Item Details</h3>
            <div className="item-detail">Target Quantity: 4/5</div>
            <div className="item-detail">Price/Item: $10.00</div>
            <div className="item-detail">Item Link: exampleurl.com/item</div>
            <div className="item-detail">
              Description:
              {' '}
              <br />
              {' '}
              Beautiful fairy lights you can use for your dorm
            </div>
          </div>
          <div className="group-details">
            <h3>Group Details</h3>
            <div className="size">Group size: 2</div>
            <div className="group-member">
              <ul>
                <li>
                  <div className="name">
                    {' '}
                    <i className="far fa-user-circle" />
                    {' '}
                    Anna
                  </div>
                  <div className="quantity">Quantity: 1</div>
                </li>
                <li>
                  <div className="name">
                    {' '}
                    <i className="far fa-user-circle" />
                    {' '}
                    Jeff
                  </div>
                  <div className="quantity">Quantity: 3</div>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div className="comment-section">{Comment()}</div>
        <div className="post-detail-tags">
          <div className="tags-label">Tags</div>
          <div className="tags-container">
            <div className="tag">Tag1</div>
            <div className="tag">Tag2</div>
            <div className="tag">Tag3</div>
            <div className="tag">Tag4</div>
            <div className="tag">Tag5</div>
            <div className="tag">Tag6</div>
          </div>
        </div>
        <div className="buttons-container">
          <div className="desired-quantity"><input data-testid="quantity-input" placeholder="desired quantity" onChange={(e) => setDesiredQuantity(e.target.value)} /></div>
          <button className="create" type="button" onClick={joinGroup}>
            Join
          </button>
          <button className="cancel" type="button" onClick={back}>
            Back
          </button>
        </div>
      </div>
    </div>
  );
}

export default PostDetails;
