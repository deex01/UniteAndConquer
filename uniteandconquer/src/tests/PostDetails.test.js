/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import PostDetails from '../components/PostDetails';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('post details title header', () => {
  render(<PostDetails />);
  const titleElement = screen.getByText('Item Name');
  expect(titleElement).toBeInTheDocument();
});

test('go back and not join group', () => {
  render(<PostDetails />);
  fireEvent.click(screen.getByText('Back'));
});

test('join group', () => {
  render(<PostDetails />);
  fireEvent.click(screen.getByText('Join'));
});

test('inputting desired quantity before joining', () => {
  render(<PostDetails />);
  const input = screen.getByTestId('quantity-input');
  fireEvent.change(input, { target: { value: '2' } });
  expect(input.value).toBe('2');
});

test('snapshot post-details', () => {
  const component = renderer.create(<PostDetails />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
