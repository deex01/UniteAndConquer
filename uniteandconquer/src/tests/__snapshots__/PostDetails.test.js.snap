// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot post-details 1`] = `
<div
  className="post-details-page"
>
  <div
    className="sidebar"
  >
    <div
      className="greeting"
    >
      <i
        className="far fa-user-circle fa-2x"
      />
       
      Hi, Jeremy
    </div>
    <div
      className="profile"
    />
    <div
      className="chat"
    />
    <div
      className="logout"
    />
    <div
      className="ad"
    />
  </div>
  <div
    className="post-details-container"
  >
    <div
      className="item-name"
    >
      <h1>
        Item Name
      </h1>
    </div>
    <div
      className="post-details-content"
    >
      <div
        className="item-details"
      >
        <h3>
          Item Details
        </h3>
        <div
          className="item-detail"
        >
          Target Quantity: 4/5
        </div>
        <div
          className="item-detail"
        >
          Price/Item: $10.00
        </div>
        <div
          className="item-detail"
        >
          Item Link: exampleurl.com/item
        </div>
        <div
          className="item-detail"
        >
          Description:
           
          <br />
           
          Beautiful fairy lights you can use for your dorm
        </div>
      </div>
      <div
        className="group-details"
      >
        <h3>
          Group Details
        </h3>
        <div
          className="size"
        >
          Group size: 2
        </div>
        <div
          className="group-member"
        >
          <ul>
            <li>
              <div
                className="name"
              >
                 
                <i
                  className="far fa-user-circle"
                />
                 
                Anna
              </div>
              <div
                className="quantity"
              >
                Quantity: 1
              </div>
            </li>
            <li>
              <div
                className="name"
              >
                 
                <i
                  className="far fa-user-circle"
                />
                 
                Jeff
              </div>
              <div
                className="quantity"
              >
                Quantity: 3
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="comment-section"
    >
      <div
        className="comments"
      >
        <div
          className="comment"
        >
          <div
            className="comment-img"
          >
            <i
              className="far fa-user-circle "
            />
          </div>
          <div
            className="comment-text"
          >
            <div
              className="comment-name"
            >
              user1
            </div>
            <div
              className="comment-content"
            >
              I loved using this item
            </div>
          </div>
        </div>
        <div
          className="comment"
        >
          <div
            className="comment-img"
          >
            <i
              className="far fa-user-circle "
            />
          </div>
          <div
            className="comment-text"
          >
            <div
              className="comment-name"
            >
              user2
            </div>
            <div
              className="comment-content"
            >
              I wonder if I will need this
            </div>
          </div>
        </div>
        <div
          className="add-comment"
        >
          <i
            className="far fa-user-circle"
          />
          <input
            className="input-comment"
            data-testid="comment-input"
            onChange={[Function]}
          />
          <button
            className="submit-comment"
            onClick={[Function]}
            type="button"
          >
            <div
              className="button-text"
            >
              Comment
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="post-detail-tags"
    >
      <div
        className="tags-label"
      >
        Tags
      </div>
      <div
        className="tags-container"
      >
        <div
          className="tag"
        >
          Tag1
        </div>
        <div
          className="tag"
        >
          Tag2
        </div>
        <div
          className="tag"
        >
          Tag3
        </div>
        <div
          className="tag"
        >
          Tag4
        </div>
        <div
          className="tag"
        >
          Tag5
        </div>
        <div
          className="tag"
        >
          Tag6
        </div>
      </div>
    </div>
    <div
      className="buttons-container"
    >
      <div
        className="desired-quantity"
      >
        <input
          data-testid="quantity-input"
          onChange={[Function]}
          placeholder="desired quantity"
        />
      </div>
      <button
        className="create"
        onClick={[Function]}
        type="button"
      >
        Join
      </button>
      <button
        className="cancel"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
